frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -610,3 +598,14 @@ frappe.views.QueryReport = class QueryReport extends frappe.views.BaseList {
 		return this.get_filter_values;
 	}
 };
+
+Object.defineProperty(frappe, 'query_report_filters_by_name', {
```

Comment:
cleaner API ? 

`frappe.provide('frappe.query_report_filters_by_name')`

Reply:
I think it will only create an empty Object.

Here I wanted that property to have only getter option and not the setter (to avoided unwanted mutation).

Also, it will always get the data based on the current report filter.